test/functional/p2p_disconnect_ban.py

```diff
@@ -70,6 +70,15 @@ def run_test(self):
         self.nodes[1].setmocktime(old_time + 3)
         assert_equal(len(self.nodes[1].listbanned()), 3)
 
+        self.log.info("Test ban_duration and time_remaining")
+        for ban in self.nodes[1].listbanned():
+            if ban["address"] in ["127.0.0.0/32", "127.0.0.0/24"]:
+                assert_equal(ban["ban_duration"], 86400)
+                assert_equal(ban["time_remaining"], 86397)
```

Comment:
```
 test  2022-01-06T13:54:00.230000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 132, in main
                                       self.run_test()
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/p2p_disconnect_ban.py", line 77, in run_test
                                       assert_equal(ban["time_remaining"], 86397)
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/util.py", line 50, in assert_equal
                                       raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
                                   AssertionError: not(86396 == 86397)
```

https://cirrus-ci.com/task/6754020390862848

Reply:
This is because the ban was set without mocktime

Reply:
```python
# Set the mocktime so we can control when bans expire
old_time = int(time.time())
self.nodes[1].setmocktime(old_time)
self.nodes[1].setban("127.0.0.0/32", "add")
self.nodes[1].setban("127.0.0.0/24", "add")
self.nodes[1].setban("192.168.0.1", "add", 1)  # ban for 1 seconds
```

instead of 

```python
self.nodes[1].setban("127.0.0.0/32", "add")
self.nodes[1].setban("127.0.0.0/24", "add")
# Set the mocktime so we can control when bans expire
old_time = int(time.time())
self.nodes[1].setmocktime(old_time)
self.nodes[1].setban("192.168.0.1", "add", 1)  # ban for 1 seconds
```

could solve it?

Reply:
Fixed in https://github.com/bitcoin/bitcoin/pull/23996